libraries/python/chat-driver/chat_driver/openai_chat_completion_driver.py

```diff
@@ -28,6 +28,51 @@
 # {"type": "json_schema", "json_schema": {"name": "test", "schema": {}}}
 
 
+class MessageHistoryProviderProtocol(Protocol):
```

Comment:
Seems like a reasonable list to start with and we can update as we find further need.  Did you check how this maps to usage in Prospector if we switch to it (the only one in this repo that I think we'd want to apply this to next)?

Reply:
Oh, these should support async since we'll be using them to get/send content from/to the workbench service.

Reply:
Are these all required or are any optional?  For example, in assistants w/ the conversation messages, I'm fine w/ get, append and extend (both point to the same there) and we _could_ support delete_all, but suspect we really don't want that.  We won't want to implement set.

Reply:
it seems this protocol should only have the get. and for message providers that support arbitrary mutation of the message history, they can implement additional methods. and those methods can be called by consumers, instead of passing those calls through the chatdriver, like is done in `clear_session_data`, `add_message`, `add_error_message`

Reply:
put another way, we should rip out all responsibility for message history management from the chatdriver. including any mutation operations.

Reply:
the LocalMessageHistoryProvider can continue to have mutation operations, for notebooks or other scenarios where you're want that sort of thing

Reply:
Ok. Trying this out now.